src/main/java/net/mcreator/generator/DefinitionsProvider.java

```diff
@@ -34,28 +35,30 @@
 
 	private static final Logger LOG = LogManager.getLogger("Definition Loader");
 
-	private final Map<ModElementType, Map<?, ?>> cache = new ConcurrentHashMap<>();
+	private final Map<String, Map<?, ?>> cache = new ConcurrentHashMap<>();

```

Comment:
Key should be ModElementType, just add equals and hashId that uses name for the source to ModElementType